app/controllers/samples_controller.rb

```diff
@@ -179,15 +179,18 @@ def index_v2
       only: [:id, :name, :host_genome_id, :project_id, :created_at, :public],
       methods: [:private_until]
     )
-    samples_visibility = get_visibility(limited_samples)
-
-    # format_samples loads a lot of information about samples
-    # There are many ways we can refactor: multiple endpoints for client to ask for the information
-    # they actually need or at least a configurable function to get only certain data
-    details_json = format_samples(limited_samples).as_json()
-    limited_samples_json.zip(details_json, samples_visibility).map do |sample, details, visibility|
-      sample[:public] = visibility
-      sample[:details] = details
+
+    # If basic requested, then don't include extra details (ex: metadata) for each sample.
+    if params[:basic].blank?
```

Comment:
I guess we are ok with the client using anything on `basic` parameter?

Reply:
I set it as `true` on the client, but it becomes a string `"true"` when passed in as a param, so I would need to do a check like `params[:basic] == "true"`. Wasn't sure what the best workaround for that was so I went with this, but let me know if there's a preferred method to handle it!

Reply:
Let's use: `basic = ActiveModel::Type::Boolean.new.cast(params[:basic])` as above. A bit more generic... 0 will be false...

Reply:
Sounds good, made that change and updated the call to have basic default to `false`.